processing/src/main/java/io/druid/query/topn/BaseTopNAlgorithm.java

```diff
@@ -74,7 +74,10 @@ public void run(
   {
     boolean hasDimValSelector = (dimValSelector != null);
 
-    final int cardinality = params.getCardinality();
+    int cardinality = params.getCardinality();
```

Comment:
Why this change should be a part of this PR? cardinality < 0 probably means that completely different approach should be taken to compute topN, setting just to `Integer.MAX_VALUE` may result in a lot of excessive work and allocations in `run()` method.